packages/next/next-server/server/config-utils.ts

```diff
@@ -23,6 +25,8 @@ export async function shouldLoadWithWebpack5(
   phase: string,
   dir: string
 ): Promise<boolean> {
+  await loadEnvConfig(dir, phase === PHASE_DEVELOPMENT_SERVER, Log)
```

Comment:
This should be handled in `loadConfig` of `next-server/server/config.ts`

Reply:
This is run inside of a separate process so we need to load it separately here